app/lib/meadow/utils/map.ex

```diff
@@ -0,0 +1,28 @@
+defmodule Meadow.Utils.Map do
+  @moduledoc """
+  Functions for manipulating maps (or structs)
+  """
+
+  @doc """
+  Replaces empty map values with nil
+
+  Examples:
+
+    iex> nillify_empty(%{a: %{}, b: %{}, c: %{}})
+    %{a: nil, b: nil, c: nil}
+
+    iex> nillify_empty(%{a: "value", b: %{}, c: "other value"})
+    %{a: "value", b: nil, c: "other value"}
+
+    iex> nillify_empty("")
+    ** (FunctionClauseError) no function clause matching in Meadow.Utils.Map.nillify_empty/1
+  """
+  def nillify_empty(map) when is_map(map) do
```

Comment:
Does this work recursively? Does it need to? Does it work on lists of maps? Will we ever run into that?

Reply:
It did work recursively, but now doesn't 😄 Want me to remove the guard clause?

Reply:
I see this discussion is happening elsewhere. I'll join in there.